### PR TITLE
80 feature instancio implementation

### PR DESCRIPTION
**Pull request by JesseMattson**

### CODE REVIEW

File: src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java [hunk 1]
```diff
@@ -150,6 +150,7 @@ public void updatePet_whenPetExists_returnsUpdatedPet() {
         final var fakeOriginalPet = fakeDataGenerator.generateFakePet();
         final var fakeUpdatedPet = fakeDataGenerator.generateFakePet();
         fakeUpdatedPet.setId(fakeOriginalPet.getId());
+        fakeUpdatedPet.setAccount(fakeOriginalPet.getAccount());
```
[the remark this answers is not in the document]
Comment (RandallKeur) [reply src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java:153]: Only alternative I can think of is if we overload the generateFakePet method to have an overloaded version where we pass in the account. Not sure if it gets any cleaner but maybe worth refactoring and seeing how you feel about it.

File: src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java [hunk 2]
```diff
@@ -4,84 +4,85 @@
...
+    /////
+    //Instancio Models
+    /////
+    Model<Pet> petModel = Instancio.of(Pet.class)
+            .generate(field(Pet::getId), Generators::ints)
+            .generate(field(Pet::getName), Generators::string)
+            .generate(field(Pet::getType), Generators::string)
+            .generate(field(Pet::getAge), Generators::ints)
+            .generate(field(Pet::getGender), Generators::string)
+            .toModel();
```
[the remark this answers is not in the document]
Comment (JesseMattson) [reply src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java:29]: I removed this from the next commit entirely.
I thought we could using Instancio Models would be needed, but I don't think it helps for what we are doing.

File: src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (JesseMattson) [reply src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java:153]: Next commit I am using toBuilder from Lombok, which is essentially a CopyConstructor, and then setting all the fields that we care about based on the actual service.

**Comment by RandallKeur** [comment]
Looking good, I think we can keep re-factoring more tests but give them another read thru to see if we can shorten them even more

**Review by RandallKeur — APPROVED** [review]

File: pom.xml [hunk 3]
```diff
@@ -76,6 +76,14 @@
...
             <artifactId>spring-boot-starter-actuator</artifactId>
         </dependency>
+        
+
+        <dependency>
+            <groupId>org.instancio</groupId>
+            <artifactId>instancio-core</artifactId>
+            <version>5.0.1</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (RandallKeur) [inline pom.xml:86]: add documentation on our actual implementation

Comment (JesseMattson) [reply pom.xml:86]: Created a chore #83 for this so Jeffrey is unblocked for refactoring.

**Pull request merged**